packages/syft/src/syft/types/errors.py

```diff
@@ -0,0 +1,51 @@
+# stdlib
+from typing import Literal
+
+# relative
+from ..service.user.user_roles import ServiceRole
+
+SyftErrorCodes = Literal[
+    "not-found",
+    "not-permitted",
+    "database-error",
+    "serde-serialization-error",
+    "serde-deserialization-error",
+    "stash-error",
+    "worker-invalid-pool",
+    "usercode-not-approved",
+    "usercode-bad-input-policy",
+    "usercode-bad-output-policy",
+]
+
+
+class SyftError(Exception):
+    __match_args__ = "code"
+
+    def __init__(
+        self,
+        message: str,
+        code: SyftErrorCodes,
```

Comment:
I think this `code` field is a bit of an antipattern compared to the standard python way of doing it. Instead of `SyftError(code="usercode-bad-input-policy)`, we should subclass:

```
class SyftException(Exception): ...

class InputPolicyException(SyftException): ...
```

Reply:
Hey @eelcovdw! Thanks for the comment.

That's one of the approaches we've discussed so far, and is probably how we'll do it.